### PR TITLE
Add documentation for `conan cache clean`

### PR DESCRIPTION
**Pull request by RubenRBS**
Docs for https://github.com/conan-io/conan/pull/13581

### CODE REVIEW

File: reference/commands/cache.rst [hunk 1]
```diff
@@ -158,26 +158,53 @@ conan cache clean
...
+                             [pattern]
 
     Remove non-critical folders from the cache, like source, build and/or download
     (.tgz store) ones.
 
     positional arguments:
       pattern               Selection pattern for references to clean
 
-    optional arguments:
+    options:
```
Comment (czoido) [inline reference/commands/cache.rst:170]: But that's not the real output of the command, is it?

Comment (memsharded) [reply reference/commands/cache.rst:170]: I think it depends on the OS Python distro, the message is different

File: reference/commands/cache.rst [hunk 2]
```diff
@@ -158,30 +158,58 @@ conan cache clean
...
+  .. code-block:: text
+
+      $ conan cache clean * --temp
+
+
+- Remove the download folders for every ``zlib`` recipe
+
+  .. code-block:: text
+
+      $ conan cache clean zlib* --download
```
Comment (czoido) [inline reference/commands/cache.rst:210]: Maybe it's good to add an example combining several arguments ?

File: reference/commands/cache.rst [hunk 3]
```diff
@@ -158,30 +158,58 @@ conan cache clean
...
+
+
+**Examples**:
+
+
+- Remove all non-critical files:
+
+  .. code-block:: text
+
+      $ conan cache clean *
```
Comment (SSE4) [inline reference/commands/cache.rst:196]: doesn't work for me:
```
ERROR: Define at least one argument among [--source, --build, --download]
```

**Review by SSE4 — CHANGES_REQUESTED** [review]
I think it's better to quote wildcard arguments, as certain shells will expand them too early (zsh)

File: reference/commands/cache.rst [hunk 4]
```diff
@@ -158,30 +158,58 @@ conan cache clean
...
-
       -s, --source          Clean source folders
       -b, --build           Clean build folders
       -d, --download        Clean download folders
+      -t, --temp            Clean temporary folders
       -p PACKAGE_QUERY, --package-query PACKAGE_QUERY
                             Remove only the packages matching a specific query,
                             e.g., os=Windows AND (arch=x86 OR compiler=gcc)
 
+This command will remove non-critical files that Conan generates in its execution.
```
Comment (SSE4) [inline reference/commands/cache.rst:184]: what are critical and non-critical files? it's not obvious

File: reference/commands/cache.rst (hunk 3 above)
Comment (RubenRBS) [reply reference/commands/cache.rst:196]: Are you on the `release/2.0` branch? Works fine for me

File: reference/commands/cache.rst (hunk 4 above)
Comment (RubenRBS) [reply reference/commands/cache.rst:184]: Good point, let me clarify those :)

**Review by czoido — APPROVED** [review]

**Pull request merged**